ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/df_driver.py

```diff
@@ -0,0 +1,167 @@
+                        if phase == "train":
+                            distinct_categories, dfs[step_config["configurations"]["dataframeName"]] = \
+                            eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
+                            if 'distinct_categories' not in metadata:
+                                metadata["distinct_categories"] = dict()
+                            metadata["distinct_categories"][step["args"]["category"]] = distinct_categories
+                        elif phase == "inference":
+                            step["args"]["distinct_categories"] = metadata["distinct_categories"][step["args"]["category"]]
+                            _, dfs[step_config["configurations"]["dataframeName"]] = \
+                            eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
+                    else:
+                        dfs[step_config["configurations"]["dataframeName"]] = \
+                            eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
+                
+            elif step_config["stepType"] == COMBINE_DATAFRAMES:
+                if step_config["configurations"]["type"] == JOIN:
+                    left = dfs[step_config["configurations"]["joinConfiguration"]["left"]]
+                    right = dfs[step_config["configurations"]["joinConfiguration"]["right"]]
+                    dfs[step_config["configurations"]["dataframeName"]] = time_series_join([left, right])
+                
+                elif step_config["configurations"]["type"] == MERGE:
+                    left = dfs[step_config["configurations"]["mergeConfiguration"]["left"]]
+                    right = dfs[step_config["configurations"]["mergeConfiguration"]["right"]]
+                    dfs[step_config["configurations"]["dataframeName"]] = time_series_merge(left, right)
+
+        # writing it to output destination
+        final_df = dfs[contents["outputDestination"]["combinedResult"]]
+        final_df.coalesce(1).write.csv(output_path, header=True)
+
+        # writing metadata to metadata bucket during train
+        if phase == "train":
+            # metadata_connector = f'oci://{phaseInfo["connector"]["bucket"]}@{phaseInfo["connector"]["namespace"]}/{phaseInfo["connector"]["objectName"]}'
+            # with open(metadata_connector, "w") as outfile:
+            #     outfile.write(metadata)
+            object_storage_client.put_object(phaseInfo["connector"]["namespace"], phaseInfo["connector"]["bucket"], phaseInfo["connector"]["objectName"], json.dumps(metadata))
+
+    except Exception as e:
+        raise Exception(e)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--response", required=True)
+    parser.add_argument("--output_path", required=True)
```

Comment:
what about configuration file path?

Reply:
`response` is the config content, and it's actually parsed and passed by OCI Fn. You can see https://bitbucket.oci.oraclecorp.com/users/shujchen/repos/shujie-draft/browse/start_dataflow_run.py for function code